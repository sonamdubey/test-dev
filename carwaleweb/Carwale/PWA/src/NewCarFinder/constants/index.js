/**
 * API URL's
 */
export const SUGGESTED_BUDGET_ENDPOINT = "/api/newcarfinder/budget/"
export const NEWCAR_SEARCH_ENDPOINT = "/api/newcarfinder/search/"
export const SUGGESTED_BODYTYPE_ENDPOINT = "/api/newcarfinder/bodytype/"
export const FUELTYPE_ENDPOINT = "/api/newcarfinder/fueltype/"
export const MAKE_ENDPOINT = "/api/newcarfinder/filters/make/"
export const VERSION_CITY_PRICE_ENDPOINT = "/api/prices/version/"
export const DEFAULT_FILTERS_ENDPOINT = "/api/filters/all/"
export const FILTER_SCREEN_ENDPOINT = "/api/newcarfinder/screen/"

/**
 * PAGE URL's
 */
export const NEWCARFINDER_ENDPOINT = '/find-car/'
export const NEWCARFINDER_FILTERS_ENDPOINT = '/find-car/filters/'
export const NEWCARFINDER_RESULTS_ENDPOINT = '/find-car/results/'
export const NEWCARFINDER_FILTERS_BODYTYPE_ENDPOINT = NEWCARFINDER_FILTERS_ENDPOINT+"body-type/"
export const NEWCARFINDER_FILTERS_FUELTYPE_ENDPOINT = NEWCARFINDER_FILTERS_ENDPOINT+"fuel-type/"
export const NEWCARFINDER_FILTERS_MAKE_ENDPOINT = NEWCARFINDER_FILTERS_ENDPOINT+"make/"
export const NEWCARFINDER_FILTERS_BUDGET_ENDPOINT = NEWCARFINDER_FILTERS_ENDPOINT+"budget/"
/**
 * SEO Pages URL's
 */
export const NEWCARFINDER_SEO_SINGLE_DIMENSION_ENDPOINT = '/best-([a-zA-Z]+)-cars-in-india/'

/**
 * NCF CONSTANTS
*/
export const BUDGET_SLIDER_MIN = 200000
export const BUDGET_SLIDER_MAX = 10000000
export const CATEGORY_NAME = "NCFPWA"
export const PAGE_VIEW_CATEGORY = "PageViews"
export const SEO_TYPE = {
    SEO_SINGLE: 'SEO_SINGLE'
}
