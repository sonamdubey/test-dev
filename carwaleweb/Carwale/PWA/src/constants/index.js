/**
 * API URL's
 */
export const CITY_AUTOCOMPLETE_URL = "/api/v2/autocomplete/city/"
export const AREA_AUTOCOMPLETE_URL = "/api/v2/autocomplete/areas/"


/**
 * UNIQUE ID's
 */
export const UNIVERSAL_ANALYTICS_ID = "UA-337359-1"


/**
 * App download links
 */
export const APP_STORE_LINK = 'https://itunes.apple.com/in/app/carwale/id910137745?mt=8'
export const PLAY_STORE_FOOTER_LINK = 'https://play.google.com/store/apps/details?id=com.carwale&referrer=utm_source%3DCarWaleMsite%26utm_medium%3DFooter%26utm_campaign=CarWale%2520MobilesiteFooter'
export const PLAY_STORE_NAV_DRAWER_LINK = 'https://play.google.com/store/apps/details?id=com.carwale&referrer=utm_source%3DCarWaleMsite%26utm_medium%3DMenuDrawer%26utm_campaign=CarWale%2520MobilesiteMenuDrawer'


export const desktopBreakPoint = 780
