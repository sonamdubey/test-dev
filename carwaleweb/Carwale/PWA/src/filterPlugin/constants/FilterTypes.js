const filterTypes = {
    budget : 1,
    fuelType : 2,
    bodyType : 3,
    make : 4,
    transmissionType : 5
}

export default filterTypes
