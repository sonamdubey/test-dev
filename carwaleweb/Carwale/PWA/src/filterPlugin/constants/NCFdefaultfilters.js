import snapPoints from '../utils/rheostat/constants/budgetSnapPoints'

const ncfDefaultFilters = () => {
    return {
        location: {
            cityId: -1,
            cityName: "",
            userConfirmed: false
        },
        newCarFinder: {
            budget: {
                displayName: "Budget",
                slider: {
                    min: 200000,
                    max: 10000000,
                    values: [
                        0
                    ],
                    userChange: false,
                    snapPoints
                },
                inputBox: {
                    value: 0
                }
            },
            bodyType: {
                displayName: "Body Type",
                data: [
                    {
                        id: 6,
                        name: 'SUV/MUV',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/suv_clr.svg',
                        lineIcon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/suv.svg',
                        isSelected: false
                    },
                    {
                        id: 3,
                        name: 'Hatchback',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/hatchback_clr.svg',
                        lineIcon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/hatchback.svg',
                        isSelected: false
                    },
                    {
                        id: 1,
                        name: 'Sedan',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/sedan_clr.svg',
                        lineIcon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/sedan.svg',
                        isSelected: false
                    },
                    {
                        id: 5,
                        name: 'Convertible',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/convertible_clr.svg',
                        lineIcon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/convertible.svg',
                        isSelected: false
                    },
                    {
                        id: 2,
                        name: 'Coupe',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/coupe_clr.svg',
                        lineIcon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/coupe.svg',
                        isSelected: false
                    },
                    {
                        id: 10,
                        name: 'Compact Sedan',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/compactsedan_clr.svg',
                        lineIcon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/compactsedan.svg',
                        isSelected: false
                    },
                    {
                        id: 4,
                        name: 'Minivan/Van',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/van_clr.svg',
                        lineIcon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/van.svg',
                        isSelected: false
                    },
                    {
                        id: 8,
                        name: 'Station Wagon',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/wagon_clr.svg',
                        lineIcon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/wagon.svg',
                        isSelected: false
                    },
                    {
                        id: 7,
                        name: 'Truck',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/truck_clr.svg',
                        lineIcon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/truck.svg',
                        isSelected: false
                    }
                ]
            },
            fuelType: {
                displayName: "Fuel Type",
                data: [
                    {
                        id: 1,
                        name: 'Petrol',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/fuel/svg/petrol.svg',
                        isSelected: false
                    },
                    {
                        id: 2,
                        name: 'Diesel',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/fuel/svg/diesel.svg',
                        isSelected: false
                    },
                    {
                        id: 3,
                        name: 'CNG',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/fuel/svg/cng.svg',
                        isSelected: false
                    },
                    {
                        id: 5,
                        name: 'Electric',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/fuel/svg/electric.svg',
                        isSelected: false
                    }
                ]
            },
            make: {
                displayName: "Make",
                data: [
                    {
                        "makeId": 10,
                        "makeName": "Maruti Suzuki",
                        "isSelected": false
                    },
                    {
                        "makeId": 8,
                        "makeName": "Hyundai",
                        "isSelected": false
                    },
                    {
                        "makeId": 7,
                        "makeName": "Honda",
                        "isSelected": false
                    },
                    {
                        "makeId": 17,
                        "makeName": "Toyota",
                        "isSelected": false
                    },
                    {
                        "makeId": 16,
                        "makeName": "Tata",
                        "isSelected": false
                    },
                    {
                        "makeId": 9,
                        "makeName": "Mahindra",
                        "isSelected": false
                    },
                    {
                        "makeId": 5,
                        "makeName": "Ford",
                        "isSelected": false
                    },
                    {
                        "makeId": 1,
                        "makeName": "BMW",
                        "isSelected": false
                    },
                    {
                        "makeId": 15,
                        "makeName": "Skoda",
                        "isSelected": false
                    },
                    {
                        "makeId": 11,
                        "makeName": "Mercedes-Benz",
                        "isSelected": false
                    },
                    {
                        "makeId": 20,
                        "makeName": "Volkswagen",
                        "isSelected": false
                    },
                    {
                        "makeId": 18,
                        "makeName": "Audi",
                        "isSelected": false
                    },
                    {
                        "makeId": 4,
                        "makeName": "Fiat",
                        "isSelected": false
                    },
                    {
                        "makeId": 12,
                        "makeName": "Mitsubishi",
                        "isSelected": false
                    },
                    {
                        "makeId": 45,
                        "makeName": "Renault",
                        "isSelected": false
                    },
                    {
                        "makeId": 21,
                        "makeName": "Nissan",
                        "isSelected": false
                    },
                    {
                        "makeId": 23,
                        "makeName": "Land Rover",
                        "isSelected": false
                    },
                    {
                        "makeId": 44,
                        "makeName": "Jaguar",
                        "isSelected": false
                    },
                    {
                        "makeId": 19,
                        "makeName": "Porsche",
                        "isSelected": false
                    },
                    {
                        "makeId": 54,
                        "makeName": "Ssangyong",
                        "isSelected": false
                    },
                    {
                        "makeId": 37,
                        "makeName": "Volvo",
                        "isSelected": false
                    },
                    {
                        "makeId": 51,
                        "makeName": "Mini",
                        "isSelected": false
                    },
                    {
                        "makeId": 22,
                        "makeName": "Bentley",
                        "isSelected": false
                    },
                    {
                        "makeId": 34,
                        "makeName": "Lexus",
                        "isSelected": false
                    },
                    {
                        "makeId": 50,
                        "makeName": "Force Motors",
                        "isSelected": false
                    },
                    {
                        "makeId": 33,
                        "makeName": "Ferrari",
                        "isSelected": false
                    },
                    {
                        "makeId": 43,
                        "makeName": "Jeep",
                        "isSelected": false
                    },
                    {
                        "makeId": 49,
                        "makeName": "Aston Martin",
                        "isSelected": false
                    },
                    {
                        "makeId": 47,
                        "makeName": "Bugatti",
                        "isSelected": false
                    },
                    {
                        "makeId": 56,
                        "makeName": "Datsun",
                        "isSelected": false
                    },
                    {
                        "makeId": 61,
                        "makeName": "Isuzu",
                        "isSelected": false
                    },
                    {
                        "makeId": 30,
                        "makeName": "Lamborghini",
                        "isSelected": false
                    },
                    {
                        "makeId": 36,
                        "makeName": "Maserati",
                        "isSelected": false
                    },
                    {
                        "makeId": 25,
                        "makeName": "Rolls-Royce",
                        "isSelected": false
                    },
                    {
                        "makeId": 66,
                        "makeName": "DC",
                        "isSelected": false
                    },
                    {
                        "makeId": 70,
                        "makeName": "Kia",
                        "isSelected": false
                    }
                ]
            },
            transmissionType: {
                displayName: "Transmission Type",
                data: [
                    {
                        id: 1,
                        name: 'Automatic',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/transmission/svg/automatic_clr.svg?v=1',
                        isSelected: false
                    },
                    {
                        id: 2,
                        name: 'Manual',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/transmission/svg/manual_clr.svg?v=1',
                        isSelected: false
                    },
                    {
                        id: 3,
                        name: 'Semi - Automatic',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/transmission/svg/semiautomatic_clr.svg?v=1',
                        isSelected: false
                    }
                ]
            },
            seats: {
                displayName: "Seating Capacity",
                data: [
                    {
                        id: 5,
                        name: '5',
                        isSelected: false
                    },
                    {
                        id: 6,
                        name: '6',
                        isSelected: false
                    },
                    {
                        id: 7,
                        name: '7',
                        isSelected: false
                    },
                    {
                        id: 8,
                        name: '8',
                        isSelected: false
                    },
                    {
                        id: 9,
                        name: '8+',
                        isSelected: false
                    }
                ]
            }
        }
    }
}

export default ncfDefaultFilters
