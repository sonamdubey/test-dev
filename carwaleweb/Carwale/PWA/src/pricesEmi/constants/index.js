
/**
 * PAGE URL's
 */

import {
	isDesktop
} from '../../utils/Common';

export const EMIPRICES_ENDPOINT = '/emi-prices/';
export const trackingCategory = isDesktop ? "Desktop_EMICalculatorPage" : "MSite_EMICalculatorPage";
