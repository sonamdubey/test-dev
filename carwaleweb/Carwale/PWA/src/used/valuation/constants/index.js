/**
 * PAGE URL's
 */
export const VALUATION_ENDPOINT = '/used/carvaluation/';

export const BUY_CAR_ID = 1
export const SELL_CAR_ID = 2

// kilometer
export const MIN_KILOMETER_DRIVEN = 1
export const MAX_KILOMETER_DRIVEN = 999999

// api end points
export const MAKE_END_POINT='/webapi/carmakesdata/getcarmakes/'
export const MODEL_END_POINT='/webapi/carmodeldata/GetCarModelsByType/'
export const VERSION_END_POINT='/webapi/carversionsdata/GetCarVersions/'
export const VALUATION_HTML_END_POINT='/m/used/valuation/v1/report/'
export const CITY_MASKING_NAME_END_POINT='/api/city/'
export const VERSION_DETAILS_END_POINT='/webapi/CarVersionsData/GetCarDetailsByVersionId/'
export const MODEL_DETAILS_END_POINT = '/webapi/CarModelData/GetCarDetailsByModelId/'

//manufacturing year
export const DEFAULT_MANUFACTURING_YEAR = 1997
export const MIN_MANUFACTURING_YEAR = 1998
export const MAX_MANUFACTURING_YEAR = new Date().getFullYear()

//Report CTA button text
export const BUY_CAR_BUTTON_TEXT = 'View Similar Cars'
export const SELL_CAR_BUTTON_TEXT = 'Sell Your Car'
