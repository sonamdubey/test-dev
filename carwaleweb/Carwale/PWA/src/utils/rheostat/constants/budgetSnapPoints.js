const budgetArray = [
	{
		step: 0.5,
		start: 200000,
		end: 1300000
	},
	{
		step: 1,
		start: 1300000,
		end: 2800000
	},
	{
		step: 2,
		start: 2800000,
		end: 3400000
	},
	{
		step: 5,
		start: 3500000,
		end: 8000000
	},
	{
		step: 10,
		start: 8000000,
		end: 10000000
	}
]

const snapPoints = []

for (let i = 0; i < budgetArray.length; i++) {
    let budget = budgetArray[i]

    for (let j = budget.start; j <= budget.end; j += budget.step * 100000) {
        if (!(snapPoints.indexOf(j) > -1)) {
            snapPoints.push(j)
        }
    }
}

export default snapPoints
