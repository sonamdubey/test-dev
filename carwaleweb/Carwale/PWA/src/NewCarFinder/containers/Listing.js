import React from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import FiltersScreen from '../../filterPlugin/containers/FiltersScreen'
import FilterContainer from './FilterContainer'
import ModelList from './ModelList'
import CompareCars from './CompareCars'
import Heading from '../components/Heading'
import { deserialzeQueryStringToObject, serialzeObjectToQueryString } from '../../utils/Common'
import ncfDefaultFilters from '../../filterPlugin/constants/NCFdefaultfilters'

import { reduceToSelectionObject, mergeQStoStore, getNCFFilterParams, getFilterStoreObject, modifySearchParamsWithStoredFilters } from '../../filterPlugin/selectors/FilterPluginSelectors'

import { setCurrentScreen } from '../actionCreators/Filter';
import { NEWCARFINDER_RESULTS_ENDPOINT } from '../constants/index';
import {setFilters} from '../../filterPlugin/actionCreators/FiltersScreen'
import {
	showFooter
} from '../../actionCreators/Footer'
import {
	showHeader
} from '../../actionCreators/Header'
import {
	showShortlistIcon,makeHeaderSticky
} from '../actionCreators/HeaderWrapper'
import {
	hideShortlistPopup
} from '../actionCreators/Shortlist'
import ShortlistPopup from '../components/ShortlistPopup'
import Footer from '../../containers/Footer'
import { unlockScroll, lockScroll } from '../../utils/ScrollLock'
import { trackCustomData } from '../../utils/cwTrackingPwa';
import {CATEGORY_NAME} from '../constants/index'
import { Helmet } from 'react-helmet'
import { getTitle, getHeading, getDimensions } from '../utils/seo-tags'
/**
 * This class representing the container component
 * for listing screen
 *
 * @class Listing
 * @extends {React.Component}
 */
class Listing extends React.Component {
	constructor(props) {
		super(props)
		if (this.props.currentScreen !== this.props.maxScreens && this.props.currentScreen > -1) {
			const filterParams = getNCFFilterParams(this.props.state)
			const qsFromStore = '?' + serialzeObjectToQueryString(filterParams)
			if(qsFromStore!==this.props.location.search){
				this.props.history.replace((NEWCARFINDER_RESULTS_ENDPOINT + qsFromStore))
			}
		}
		this.renderResults = !this.props.shortlistCars.active
		this.shouldBeAtPageTop = true
		}

	componentWillMount(){
        this.props.showHeader()
		this.props.showShortlistIcon()
		this.props.makeHeaderSticky()
	}

	componentDidMount() {
		this.props.setCurrentScreen(this.props.maxScreens)
		trackCustomData(CATEGORY_NAME, "ListingScreenImpression", "NA", false)
	}
	componentWillUpdate(nextProps){
		if(nextProps.shortlistCars.active){
			lockScroll()
		}
		this.shouldBeAtPageTop = nextProps.shortlistCars.active === this.props.shortlistCars.active
	}
	componentDidUpdate(){
		if(this.props.shortlistCars.count == 0 && !this.props.shortlistCars.active){
			unlockScroll()
		}
	}
	shouldComponentUpdate(nextProps, nextState) {
		if(nextProps.location.search != this.props.location.search) {
			return true
		}
		if(nextProps.shortlistCars.active != this.props.shortlistCars.active) {
			return true
		}
		if(nextProps.isSEOPage != this.props.isSEOPage) {
			return true
		}
		return false
	}

	handleShortlistBack = () => {
		this.props.hideShortlistPopup()
		unlockScroll()
	}

	renderMainListingPage (searchParams,seoData){
		if(this.renderResults){
			return (
				<ModelList key="mainListing" isSEOPage={this.props.isSEOPage}  pagename="mainListing" searchParams={searchParams} seoData={seoData}/>
			)
		}
		else{
			this.renderResults = true
		}
	}
	renderHelmetComponent (seoData) {
		return (
			<Helmet>
				<title>{seoData.title}</title>
				<link rel="canonical" href={seoData.url} />
			</Helmet>
		)
	}
	getSEOData (searchParams) {
		let dimension = getDimensions(searchParams)
		let title = getTitle(searchParams,dimension)
		let heading = getHeading(searchParams,dimension,false)
		let url = window.location.href
		let h1 = getHeading(searchParams,dimension,true)
		return {title , heading, url, dimension, h1 }
	}
	render() {
		if(this.shouldBeAtPageTop){
			window.scrollTo(0, 0)
		}
		const ncfDefaultFiltersClone = ncfDefaultFilters()
		const searchParams = deserialzeQueryStringToObject(this.props.location.search)

		if(this.props.shortlistCars.removedModelIds != undefined && this.props.shortlistCars.removedModelIds.length){
			searchParams.removedModelIds =this.props.shortlistCars.removedModelIds
		}

		let seoData = null,
			headingComponent = null,
			helmetComponent = null,
			modifiedSearchParams = {
				... {},
				... searchParams
			}
		if(this.props.isSEOPage) {
			seoData = this.getSEOData (searchParams)
			helmetComponent = this.renderHelmetComponent(seoData)
			headingComponent =	<Heading heading={seoData.h1} />
			modifiedSearchParams = modifySearchParamsWithStoredFilters(modifiedSearchParams, this.props.storedFilters)
		}
		else{
			this.props.setFilters(getFilterStoreObject(searchParams))
			helmetComponent = (
				<Helmet>
					<title>New Car Finder - CarWale.com</title>
				</Helmet>
			)
		}
		const defaultFilters = mergeQStoStore(modifiedSearchParams, ncfDefaultFiltersClone)
		const selectionObject = reduceToSelectionObject(defaultFilters)

		return (
			<div className="listing-screen">
				{helmetComponent}
				{headingComponent}
				{this.renderMainListingPage(searchParams,seoData)}
				<CompareCars />
				<FilterContainer
					withCompare={this.props.state.compareCars.active}
					searchParams={searchParams} />
				<FiltersScreen
					defaultFilters={defaultFilters}
					selectionObject={selectionObject}
					trackingCategory={CATEGORY_NAME}
					location={this.props.location}
					history={this.props.history}
					match={this.props.match}
				/>
				<ShortlistPopup status={this.props.shortlistCars.active} handleBack={this.handleShortlistBack} cityId={searchParams.cityId}/>
				<Footer />
			</div>
		)
	}
}

const mapStateToProps = (state) => {
	const filterScreens = state.newCarFinder.filter.filterScreens
	const currentScreen = filterScreens.currentScreen
	const maxScreens = filterScreens.screenOrder.length
	const city = state.location
	const {
			shortlistCars
		} = state.newCarFinder
	const {
		filters: storedFilters
	} = state.filtersScreen
	return {
		currentScreen,
		maxScreens,
		state,
		shortlistCars,
		city,
		storedFilters
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		setCurrentScreen: bindActionCreators(setCurrentScreen, dispatch),
		showFooter: bindActionCreators(showFooter, dispatch),
		showHeader: bindActionCreators(showHeader, dispatch),
		hideShortlistPopup: bindActionCreators(hideShortlistPopup, dispatch),
		showShortlistIcon: bindActionCreators(showShortlistIcon, dispatch),
		makeHeaderSticky: bindActionCreators(makeHeaderSticky, dispatch),
		setFilters: bindActionCreators(setFilters, dispatch)
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(Listing)
