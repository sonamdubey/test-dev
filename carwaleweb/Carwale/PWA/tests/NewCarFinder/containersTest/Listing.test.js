import React from 'react'
import { Redirect } from 'react-router-dom'
import Listing from '../../../src/NewCarFinder/containers/Listing'
import { NEWCARFINDER_RESULTS_ENDPOINT } from '../../../src/NewCarFinder/constants/index';

describe('Tests for Listing component', () => {
    let state
    let props
    let store
    let enzymeWrapper

    const Wrapper = () => {
        if(!enzymeWrapper){
            store = mockStore(state)
            enzymeWrapper = shallowWithStore(<Listing {...props}/>, store)

        }
        return enzymeWrapper
    }

    beforeEach( () => {
        props = undefined
        state = undefined
        enzymeWrapper = undefined
        store = undefined
    })

    it('should render snapshot', () => {
        props = {
            history: {
                replace:jest.fn()
            },
            location:{
                pathname: NEWCARFINDER_RESULTS_ENDPOINT,
                search: '?cityId=1&budget=1465060&bodyStyleIds=3%2C1'
            }
        }
        state = getMockInitialStore()
        const wrapper = Wrapper()
        expect(wrapper).toMatchSnapshot()
      });

      describe('ShouldComponentUpdate', () => {

        it('should return true if querystring is different', () => {
            props = {
                location:{
                    pathname: NEWCARFINDER_RESULTS_ENDPOINT,
                    search: '?cityId=1&budget=1465060&bodyStyleIds=3%2C1'
                }
            }

            let nextProps = {
                location:{
                    pathname: NEWCARFINDER_RESULTS_ENDPOINT,
                    search: '?cityId=10&budget=1465060&bodyStyleIds=3%2C1'
                }
            }
            state = getMockInitialStore()
            const wrapper = Wrapper()
            expect(wrapper.dive().instance().shouldComponentUpdate(nextProps)).toBeTruthy()

        })

        it('should return false if querystring is same', () => {
            props = {
                location:{
                    pathname: NEWCARFINDER_RESULTS_ENDPOINT,
                    search: '?cityId=1&budget=1465060&bodyStyleIds=3%2C1'
                }
            }

            let nextProps = {
                location:{
                    pathname: NEWCARFINDER_RESULTS_ENDPOINT,
                    search: '?cityId=1&budget=1465060&bodyStyleIds=3%2C1'
                }
            }
            state = getMockInitialStore()
            const wrapper = Wrapper()
            expect(wrapper.instance().shouldComponentUpdate(nextProps)).toBeFalsy()

        })
      })
})
