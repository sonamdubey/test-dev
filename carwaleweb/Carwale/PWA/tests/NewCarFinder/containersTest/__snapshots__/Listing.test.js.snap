// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Listing component should render snapshot 1`] = `
<Listing
  history={
    Object {
      "replace": [MockFunction],
    }
  }
  location={
    Object {
      "pathname": "/find-car/results/",
      "search": "?cityId=1&budget=1465060&bodyStyleIds=3%2C1",
    }
  }
  maxScreens={4}
  setCurrentScreen={[Function]}
  showFooter={[Function]}
  showHeader={[Function]}
  state={
    Object {
      "location": Object {
        "cityId": 1,
        "cityname": "Mumbai",
        "userConfirmed": false,
      },
      "newCarFinder": Object {
        "bodyType": Object {
          "data": Array [],
          "displayName": "Body Type",
          "isFetching": false,
        },
        "budget": Object {
          "displayName": "Budget",
          "inputBox": Object {
            "value": 300000,
          },
          "slider": Object {
            "max": 10000000,
            "min": 200000,
            "snapPoints": Array [
              200000,
              250000,
              300000,
              350000,
              400000,
              450000,
              500000,
              550000,
              600000,
              650000,
              700000,
              750000,
              800000,
              850000,
              900000,
              950000,
              1000000,
              1050000,
              1100000,
              1150000,
              1200000,
              1250000,
              1300000,
              1400000,
              1500000,
              1600000,
              1700000,
              1800000,
              1900000,
              2000000,
              2100000,
              2200000,
              2300000,
              2400000,
              2500000,
              2600000,
              2700000,
              2800000,
              3000000,
              3200000,
              3400000,
              3500000,
              4000000,
              4500000,
              5000000,
              5500000,
              6000000,
              6500000,
              7000000,
              7500000,
              8000000,
              9000000,
              10000000,
            ],
            "userChange": false,
            "values": Array [
              300000,
            ],
          },
        },
        "filter": Object {
          "filterScreens": Object {
            "screenOrder": Array [
              Object {
                "name": "BudgetFilter",
                "path": "/find-car/filters/budget",
              },
              Object {
                "name": "BodyTypeFilter",
                "path": "/find-car/filters/body-type",
              },
              Object {
                "name": "FuelTypeFilter",
                "path": "/find-car/filters/fuel-type",
              },
              Object {
                "name": "MakeFilter",
                "path": "/find-car/filters/make",
              },
            ],
          },
        },
        "fuelType": Object {
          "data": Array [],
          "displayName": "Fuel Type",
          "isFetching": false,
        },
        "make": Object {
          "data": Array [],
          "displayName": "Make",
          "isFetching": false,
        },
      },
      "toast": Object {
        "duration": 3000,
        "isVisible": false,
        "message": "",
        "style": null,
        "toastTimerId": null,
      },
    }
  }
/>
`;
